### ISSUE TITLE
OCR of binarized image does not lead to same result as OCR of input image

### ISSUE THREAD

**Issue by lloiodice** [tesseract-ocr/tesseract#1780]
### Environment

* **Tesseract Version**:  v4.0.0-beta.3-204-gf47da
* **Commit Number**: f47daace1f826f7e7be14e88a986f495f9ada542
* **Platform**: Linux ip-10-10-153-84 4.9.38-16.35.amzn1.x86_64 #1 SMP Sat Aug 5 01:39:35 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

### Current Behavior:

1. OCR a grayscale input image exporting binarized (‘Thresholded’) image 
2. OCR binarized image obtained in step 1

[image.binarized.zip](https://github.com/tesseract-ocr/tesseract/files/2197975/image.binarized.zip)

OCR of binarized image has lower accuracy then OCR of input image.
This happens on multiple files (30% of files in my test set ... which are standard emails with normal fonts) and for example when OCR-ing image.input.tif I get word 'PriceWaterhouseCoppers' with confidence 90. When I use the binarized image produced by Tesseract in the previous step, I get 2 words 'Price' and 'WaterhouseCoopers' with low confidence.

### Expected Behavior:
OCR of binarized image has same accuracy then OCR of input image.

### Additional Comments

My understanding is that tesseract very early in its process binarizes the input image and uses this internally binarized image to recognize text.

I was expecting OCR of any input image and its associated tesseract-binarized image to return the same results. I got to this observation while comparing accuracy between grayscale and B&W images. I think one of the suggestion (https://github.com/tesseract-ocr/tesseract/wiki/ImproveQuality#binarisation) is to pre-binarize the images, but I get better results when providing grayscale images … so I suspect that there might be some issue causing input B&W images to get lower accuracy. 

To easily reproduce, after running Tesseract input.image.tif input.image.tif, the output text has word PriceWaterhouseCoppers. After running Tesseract input.binarized.tif input.binarized.tif, the output text has 2 words Price and WaterhouseCoppers

Exporting the binarized image when OCR-ing the binarized image (step 2 above) leads to an identical image (as expected or hoped)

**Comment by amitdo** [comment]
I think that with v4.0, Tesseract only use the BW image for the layout analysis step.

If you give it a colored image, tesseract will convert it to a grayscale image, and it will use this grayscale image for the text recognition step.

The OCR engine can do better on grayscale then BW because when you convert to BW some of the information (=pixels) is lost.

So, it's not a bug, but rather a feature :-)

**Comment by theceday** [comment]
I ended up writing line segmentation myself, using binary images for "prediction" is not a good idea as binarization will loss/add extra pixels/data

Adaptive gaussian binaryzation seems to work well until it doesnt :)

I ended up using bradley however in order to work it properly (windows size etc), images need to be resized. (that and some other things along the line intruduces a font-size range you want to recognize whether you want it or not :))

for LSTM usage, wiki is can be a little misleading and it is not clear what is the best practices.
For example, does tesseract resize given "line" images? Or should we resize them?

First layer of tesseract is a summarizing layer for height right?
The mean value of height of images that is used in tranining should be actually important.
(i dont know what the "op" implemented there does, however a few some quick info about that would be really good)
